api/v1alpha1/addonimageset_utils.go

```diff
@@ -0,0 +1,32 @@
+package v1alpha1
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"golang.org/x/mod/semver"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// FromYAML - instantiates an AddonImageSetSpec struct from yaml data
+func (a *AddonImageSetSpec) FromYAML(data []byte) error {
```

Comment:
I noticed that we've ta est for GetSemver but not for FromYAML or ToJSON, we should add it.

Reply:
well both FromYAML and ToJSON only really use the `json` and `yaml` modules. A test would not cover anything that is owned by our codebase. These are just stylistic helpers so we can do:

```go
data, err := addonmetadata.ToJSON()

// instead of
data, err := json.Marshal(meta)
```